src/ORM/Connect/MySQLQuery.php

```diff
@@ -64,11 +64,20 @@ public function getIterator(): Traversable
                 }
                 yield $data;
             }
-            // Check for the method first since $this->handle is a mixed type
-            if (method_exists($this->handle, 'data_seek')) {
-                // Reset so the query can be iterated over again
-                $this->handle->data_seek(0);
-            }
+
+            // Reset so the query can be iterated over again
+            $this->rewind();
+        }
+    }
+
+    /**
+     * Rewind to the first row
+     */
+    public function rewind(): void
+    {
+        // Check for the method first since $this->handle is a mixed type
+        if (method_exists($this->handle, 'data_seek')) {
+            $this->handle->data_seek(0);
         }
     }
```

Comment:
This (and the same change in `MySQLStatement`)  is required so we can break a loop in `DataList::addEagerLoadedDataToParent()` and then iterate over the result again later from the beginning